core/src/main/scala/org/apache/spark/sql/delta/commands/RestoreTableCommand.scala

```diff
@@ -102,36 +103,42 @@ case class RestoreTableCommand(
             "left_anti")
           .as[AddFile]
           .map(_.removeWithTimestamp())
-          // To avoid recompute of Dataset with wide transformation by toLocalIterator
-          .cache()
 
-        try {
-          checkSnapshotFilesAvailability(deltaLog, filesToAdd, versionToRestore)
+        val ignoreMissingFiles = spark
+          .sessionState
+          .conf
+          .getConf(IGNORE_MISSING_FILES)
 
-          // Commit files, metrics, protocol and metadata to delta log
-          val metrics = computeMetrics(filesToAdd, filesToRemove, snapshotToRestore)
-          val addActions = filesToAdd.toLocalIterator().asScala
-          val removeActions = filesToRemove.toLocalIterator().asScala
-
-          txn.updateMetadata(snapshotToRestore.metadata)
-
-          commitLarge(
-            spark,
-            txn,
-            addActions ++ removeActions,
-            DeltaOperations.Restore(version, timestamp),
-            Map.empty,
-            metrics)
-        } finally {
-          filesToAdd.unpersist()
-          filesToRemove.unpersist()
+        if (!ignoreMissingFiles) {
+          checkSnapshotFilesAvailability(deltaLog, filesToAdd, versionToRestore)
         }
-      }
 
+        val metrics = withDescription("metrics") {
+          computeMetrics(filesToAdd, filesToRemove, snapshotToRestore) }
+        val addActions = withDescription("add actions") {
+          filesToAdd.toLocalIterator().asScala }
+        val removeActions = withDescription("remove actions") {
+          filesToRemove.toLocalIterator().asScala }
+
+        txn.updateMetadata(snapshotToRestore.metadata)
+        // Commit files, metrics, protocol and metadata to delta log
+        commitLarge(
+          spark,
+          txn,
+          addActions ++ removeActions,
+          DeltaOperations.Restore(version, timestamp),
+          Map.empty,
+          metrics)
+      }
       Seq.empty[Row]
     }
   }
 
+  private def withDescription[T](action: String)(f: => T): T = withStatusCode("Delta:",
```

Comment:
nit: Seems like `:` is already appended by the utility method when constructing the logging message.

Reply:
`Delta:` updated to `DELTA` (similar to other places where we use withStatusCode)